peas/weather.py

```diff
@@ -784,175 +740,55 @@ def calculate_and_set_PWM(self):
-
         # Wind (average and gusts)
         wind_speed = [x['wind_speed_KPH']
-                      for x in entries
+                      for x in self.weather_entries
                       if 'wind_speed_KPH' in x.keys()]
 
-        if len(wind_speed) == 0:
-            self.logger.debug('  UNSAFE: no wind speed readings found')
-            wind_safe = False
-            gust_safe = False
-            wind_condition = 'Unknown'
-            gust_condition = 'Unknown'
-        else:
-            start_time = entries[0]['date']
-            if type(start_time) == str:
-                start_time = date_parser(entries[0]['date'])
-
-            typical_data_interval = (end_time - start_time).total_seconds() / len(entries)
+        wind_gust = max(wind_speed)
 
-            mavg_count = int(np.ceil(120. / typical_data_interval))  # What is this 120?
-            wind_mavg = movingaverage(wind_speed, mavg_count)
-
-            # Windy?
-            if max(wind_mavg) > threshold_very_windy:
-                self.logger.debug('  UNSAFE:  Very windy in last {:.0f} min. Max wind speed {:.1f} kph'.format(
-                    safety_delay, max(wind_mavg)))
-                wind_safe = False
-            else:
-                wind_safe = True
-
-            if wind_mavg[-1] > threshold_very_windy:
-                wind_condition = 'Very Windy'
-            elif wind_mavg[-1] > threshold_windy:
-                wind_condition = 'Windy'
-            else:
-                wind_condition = 'Calm'
-            self.logger.debug('  Wind Condition: {} ({:.1f} km/h)'.format(wind_condition, wind_mavg[-1]))
-
-            # Gusty?
-            if max(wind_speed) > threshold_very_gusty:
-                self.logger.debug('  UNSAFE:  Very gusty in last {:.0f} min. Max gust speed {:.1f} kph'.format(
-                    safety_delay, max(wind_speed)))
-                gust_safe = False
-            else:
-                gust_safe = True
-
-            current_wind = current_values.get('wind_speed_KPH', 0.0)
-            if current_wind > threshold_very_gusty:
-                gust_condition = 'Very Gusty'
-            elif current_wind > threshold_gusty:
-                gust_condition = 'Gusty'
-            else:
-                gust_condition = 'Calm'
+        moving_avg_seconds = 120.
+        end_time = dt.utcnow()
+        start_time = date_parser(entries[0]['date'])
+        typical_data_interval = (end_time - start_time).total_seconds() / len(entries)
+        mavg_count = int(np.ceil(moving_avg_seconds / typical_data_interval))  # What is this 120?
+        wind_mavg = movingaverage(wind_speed, mavg_count)
 
-            self.logger.debug('  Gust Condition: {} ({:.1f} km/h)'.format(gust_condition, wind_speed[-1]))
+        wind_speed = max(wind_mavg)
 
-        return (wind_condition, wind_safe), (gust_condition, gust_safe)
+        return super()._get_wind_safety(wind_speed, wind_gust)
 
     def _get_rain_safety(self, current_values):
```

Comment:
Modify to use results from `_get_status()`.

Reply:
Modified to use the results from `_get_status()`.